### ISSUE TITLE
Magnet link to preview content and choose what to download

### ISSUE THREAD

**Issue by sdugoten** [transmission-remote-gui/transgui#1436]
1. Operating system : Windows 11
2. Transmission Deamon
3. Transmission Remote GUI  

 It seems that dragging .torrent file into the GUI would allow you to choose what file to download before start, but using magnet link would not.  Is this a overlook or this is a feature request?

**Comment by xavery** [comment]
You have no idea what's hiding behind a magnet request when you add it. The metadata (i.e. the list of all the files in a given download) is only fetched by the daemon after starting a magnetized download. What you're describing is possible with standalone .torrent files precisely because they hold the entire metadata.

**Comment by sdugoten** [comment]
> You have no idea what's hiding behind a magnet link when you add it. The metadata (i.e. the list of all the files in a given download) is only fetched by the daemon after starting a magnetized download. What you're describing is possible with standalone .torrent files precisely because they hold the entire metadata.

  Basically the Transmission will need to fetch the meta data of the magnet link and then show the pop up dialog with all the files inside the magnet link.  Most other BT client like qbittorrent, FDM5 and deluge will do that and allow you to choose which file you want to download.  Some even have "select all", "select none", "select video files" button so that you can quickly do the selection in the pop up dialog.

**Comment by sdugoten** [comment]
So, most other popular BT clients have this feature.  Some even have highly customizable filtering on the popup dialog box on magnet link and .torrent file , it will display file extension type on the top so that you can quickly choose what file type(s) you want to select without clicking every single checkbox in the file listing.

**Comment by Mrnofish** [comment]
Agreed, this would be a very useful feature and, as it has been aptly pointed out, it's been available in major Torrent clients for quite some time.

On the other hand, I do wonder whether previewing magnet links is truly a task that's purview of a humble front-end ?

As far as I can tell, the remote does not speak any part of the BitTorrent protocol, whereas it goes without saying Transmission, as a BT client, itself already does.

Transmission has a new, highly active, team, BTW, so if they haven't already tackled this issue, they might be receptive to suggestions.

**Comment by cyberhardt** [comment]
> You have no idea what's hiding behind a magnet link when you add it. The metadata (i.e. the list of all the files in a given download) is only fetched by the daemon after starting a magnetized download. What you're describing is possible with standalone .torrent files precisely because they hold the entire metadata.

and you have never used another client, apparently

**Comment by xavery** [comment]
> and you have never used another client, apparently

Thanks, a very insightful and valuable comment indeed.

**Comment by sdugoten** [comment]
wonder if this feature request will be considered in future version?

**Comment by sdugoten** [comment]
FYI, qbittorrent have already implemented the API part to show file content with magnet. they are now working on the web GUI part. It is available soon as far as I can tell on their headless version.

https://github.com/qbittorrent/qBittorrent/pull/21645

I like to use Tranmission, but I just can't live without a feature that is available almost on every single BT client out there.
